core/graph/src/main/java/uk/gov/gchq/gaffer/graph/hook/OperationAuthoriser.java

```diff
@@ -110,40 +65,53 @@ public void preExecute(final OperationChain<?> opChain, final User user) {
     }
 
     /**
-     * Set operation authorisations for a given operation class.
-     *
-     * @param opClass the operation class
-     * @param auths   the authorisations
-     */
-    public void setOpAuths(final Class<?> opClass, final Set<String> auths) {
-        opAuthsMap.put(opClass, auths);
-        allOpAuths.addAll(auths);
-    }
-
-    /**
      * Add operation authorisations for a given operation class.
      * This can be called multiple times for the same operation class and the
      * authorisations will be appended.
      *
      * @param opClass the operation class
      * @param auths   the authorisations
      */
-    public void addOpAuths(final Class<? extends Operation> opClass, final String... auths) {
-        final Set<String> opAuths = opAuthsMap.computeIfAbsent(opClass, k -> new HashSet<>());
+    public void addAuths(final Class<? extends Operation> opClass, final String... auths) {
+        final Set<String> opAuths = this.auths.computeIfAbsent(opClass, k -> new HashSet<>());
         Collections.addAll(opAuths, auths);
-        Collections.addAll(allOpAuths, auths);
+        Collections.addAll(allAuths, auths);
     }
 
-    public Set<String> getAllOpAuths() {
-        return Collections.unmodifiableSet(allOpAuths);
+    public Map<Class<?>, Set<String>> getAuths() {
+        return Collections.unmodifiableMap(auths);
+    }
+
+    public void setAuths(final Map<Class<?>, Set<String>> auths) {
+        this.auths.clear();
+        this.auths.putAll(auths);
```

Comment:
Elsewhere we are carrying out null checks before calling Collection.putAll() - should we be doing that everywhere?